niceman/tests/fixtures.py

```diff
@@ -22,6 +24,39 @@ def niceman_cfg_path(request):
     yield request.param
 
 
+
+@pytest.fixture(scope="module")
+def temp_file():
+    """Provide an empty temporary file for a test.
+
+    The fixture will create a temp file and remove it when the test is complete.
+    To add content to the temporary file, call the object in the test function
+    with the content passed as the first argument of the call.
+
+    e.g. tmp_file("my content")
+
+    Returns
+    -------
+    FixtureTempFile object
+    """
+    class FixtureTempFile:
+        def __init__(self, fid, path):
+            self.fid = fid
+            self.path = path
+        def __call__(self, content):
+            if content:
+                file = open(self.path, "w")
+                file.write(content)
+                file.close()
+        def __repr__(self):
+            return self.path
+
+    fid, path = tempfile.mkstemp()
+    temp_file = FixtureTempFile(fid, path)
+    yield temp_file
+    os.remove(path)
```

Comment:
Thoughts/my own confusions:

  * I'm probably missing something, but I don't see the advantage of
    this structure over returning the result of
    `tempfile.NamedTemporaryFile`.

  * Why open and close each call?

  * Why implement `__call__` rather than sticking to the more familiar
    `.write` for a file objects?

  * Is the `__repr__` method made use of in the current tests?

  * I would have expected this to have pytest's default function scope
    rather than module scope.

Reply:
I was focused on mkstemp() and missed NamedTemporaryFile, which is basically doing what I coded up. The __call__ was to simplify file operations but passing back the established NamedTemporaryFile makes more sense. Python noob mistake, excellent guidance. Will rework and push

Reply:
UPDATE: The simplicity of setting up a temporary file with NamedTemporaryFile makes the use of a fixture overkill. I removed the temp_file fixture in favor of using tempfile.NamedTemporaryFile directly.

Reply:
I am lost why we need another fixture in addition to existing `tmpdir` fixture in pytest (also remember we still have nose-style `@with_tempfile` with cute features suchas `content` option)

Reply:
> I am lost why we need another fixture

I believe it's already be removed in e84402f6.